cluster/v1alpha1/helpers.go

```diff
@@ -236,32 +268,54 @@ func progressivePerCluster(clusterGroupsMap clusterv1beta1.ClusterGroupsMap, len
 	}
 }
 
-func progressivePerGroup(clusterGroupsMap clusterv1beta1.ClusterGroupsMap, timeout time.Duration, statusFunc ClusterRolloutStatusFunc) RolloutResult {
-	rolloutClusters := map[string]ClusterRolloutStatus{}
-	timeoutClusters := map[string]ClusterRolloutStatus{}
+func progressivePerGroup(clusterGroupsMap clusterv1beta1.ClusterGroupsMap, timeout time.Duration, existingClusterStatus []ClusterRolloutStatus) RolloutResult {
+	rolloutClusters := []ClusterRolloutStatus{}
+	timeoutClusters := []ClusterRolloutStatus{}
+	existingClusters := make(map[string]bool)
 
-	clusterGroupKeys := clusterGroupsMap.GetOrderedGroupKeys()
+	for _, status := range existingClusterStatus {
```

Comment:
This function is to rollout by group, why go through the existingClusterStatus and put it to result? what will the returned rolloutClusters be like?

Reply:
Still have some concerns about going through the existingClusterStatus first and then going through the rest of the clusters, for both progressivePerCluster and progressivePerGroup. 

For example, the env has 100 clusters, then rollout starts, what's the expected cluster status returned by existingClusterStatus? For addon existingClusterStatus will return ToApply when rollout starts, in this case, all the clusters will be returned by GetRolloutCluster(), which doesn't make sense.

Reply:
I understand your point, same as above let me know your thoughts.

Reply:
Another gap after this change is that, in addon, one cma will have multiple placement, there's placement overlap. For example:
The config-v1 existingClusterStatus has cluster1 and config-v2 existingClusterStatus has cluster2, cluster3. For config-v1, cluster2 is not in existing cluster and it should also not be treated as added cluster and return in rollout result. 
```
  installStrategy:
    type: Placements
    placements:
    - name: placement1 // placement1 select cluster1 and cluster2
      namespace: default
      configs:
      - group: addon.open-cluster-management.io
        resource: addonhubconfigs
        name: config-v1
      rolloutStrategy:
        type: Progressive
    - name: placement2 // placement2 select cluster2 and cluster3
      namespace: default
      configs:
      - group: addon.open-cluster-management.io
        resource: addonhubconfigs
        name: config-v2
      rolloutStrategy:
        type: Progressive
```
I'm thinking should we just rollout the clusters returned by ClusterRolloutStatusFunc and put added/deleted cluster in another field.

Reply:
Addon has returned a SKIP status for cluster2 when rollout config-v1 to indicate not to rollout it. Not sure if PR can handle the case when cluster2 is not in existingClusterStatus and user want to SKIP rollout it.

Reply:
Yes, addon might need to put SKIP cluster in existingClusterStatus so that won't put it in ClustersToRollout. Sounds good.